app/order/api.go

```diff
@@ -47,11 +47,11 @@ type OrderInput struct {
 
 // OrderStatus holds the status of an Order workflow.
 type OrderStatus struct {
-	ID         string    `json:"id"`
-	CustomerID string    `json:"customerId" db:"customer_id"`
-	ReceivedAt time.Time `json:"receivedAt" db:"received_at"`
+	ID         string    `json:"id" db:"id" bson:"id"`
+	CustomerID string    `json:"customerId" db:"customer_id" bson:"customer_id"`
+	ReceivedAt time.Time `json:"receivedAt" db:"received_at" bson:"received_at"`
 
-	Status string `json:"status"`
+	Status string `json:"status" db:"status" bson:"status"`
```

Comment:
Just a note here, we should be cautious encouraging users to reuse ORM models in Temporal. We often caution users against this because, even if it seems they may be the same now, they often diverge and users cannot make them diverge gracefully. We usually encourage users have models tailored for Temporal and translate to/from their database models.

Not to say this is wrong necessarily, but an impedance mismatch across model expectations is very likely to pop up when reusing models. Sooner or later you're going to want a Temporal-specific field or a DB-specific field and this approach causes issues when that occurs.

Reply:
Adjusted to use db-specific models, which also removes the tag leak.

Reply:
Adjusted to use db specific models (which also removes the leaked struct tags).